api/envoy/extensions/transport_sockets/tls/v3/common.proto

```diff
@@ -233,8 +249,17 @@ message CertificateValidationContext {
   //
   // See :ref:`the TLS overview <arch_overview_ssl_enabling_verification>` for a list of common
   // system CA locations.
+  //
+  // If *trusted_ca* is a filesystem path, a watch will be added to the parent
+  // directory for any file moves to support rotation. This currently only
+  // applies to dynamic secrets, when the *CertificateValidationContext* is
+  // delivered via SDS.
   config.core.v3.DataSource trusted_ca = 1;
 
+  // If specified, updates of a file-based *trusted_ca* source will be triggered
+  // by this watch.
+  config.core.v3.WatchedDirectory watched_directory = 11;
```

Comment:
What is the utility of this if there is only a single file being watched? Can this be removed for now for simplicity? Or is it that this works for non-SDS? If that's the case can you beef up the docs for both fields to make this more clear? It's kind of confusing.

Reply:
This allows you to watch at arbitrary points in the file path. E.g. for `/foo/bar/baz/cert.pem`, you can watch on `/foo/bar`. I'll work on improving docs.